### PR TITLE
Scrape image:width and image:height, if available

### PR DESCRIPTION
**Pull request by pdehaan**
Still needs a bit of tweaking on page-metadata-service

### CODE REVIEW

File: parser.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ const titleRules = buildRuleset('title', [
 
 const canonicalUrlRules = buildRuleset('url', [
   ['meta[property="og:url"]', node => node.element.content],
+  ['meta[name="twitter:url"]', node => node.element.content],
```
Comment (jaredkerim) [inline parser.js]: This should be a separate PR.

File: parser.js [hunk 2]
```diff
@@ -70,16 +79,17 @@ const typeRules = buildRuleset('type', [
...
+  image_width: imageWidthRules,
   title: titleRules,
   type: typeRules,
-  url: canonicalUrlRules
+  url: canonicalUrlRules,
 };
 
 
-function getMetadata(doc, rules) {
+function getMetadata(doc, ruleSet = metadataRules) {
```
Comment (jaredkerim) [inline parser.js]: This should be a separate PR.

File: parser.js [hunk 3]
```diff
@@ -57,6 +58,14 @@ const imageRules = buildRuleset('image', [
   ['img', node => node.element.src],
 ]);
 
+const imageWidthRules = buildRuleset('image_width', [
+  ['meta[property="og:image:width"]', node => node.element.content],
```
Comment (jaredkerim) [inline parser.js]: We need tests for these new rules.

**Comment by pdehaan** [comment]
After reading http://ogp.me/#array I'm not convinced my logic would work in all/most cases.
I can definitely see a handful of cases where it'd fail spectacularly if the page returned multiple `og:image` objects, but the first one in the array didn't have a specified `width` and `height`.

**Comment by pdehaan** [comment]
Closing for now, until I'm ready to fix this properly (or we move to fastimage, per https://github.com/mozilla/page-metadata-service/issues/89)
